modules/core/src/persistence.cpp

```diff
@@ -4025,7 +4025,14 @@ static void
 icvJSONWriteReal( CvFileStorage* fs, const char* key, double value )
 {
     char buf[128];
-    icvJSONWrite( fs, key, icvDoubleToString( buf, value ));
+    int len = (int)strlen( icvDoubleToString( buf, value ) );
```

Comment:
Why do you cast the result of `strlen` (which is of unsigned type `size_t`) to an `int`? This can cause an integer overflow. Also, `strlen` does not return negative values.